nixos/modules/services/networking/openvpn.nix

```diff
@@ -39,20 +39,30 @@ let
         ${cfg.down}
       '';
 
+      hasUpScript = (cfg.up != "");
+      hasDownScript = (cfg.down != "");
+
+
+      upScriptOption = optionalString (hasUpScript || cfg.updateResolvConf) "up ${pkgs.writeShellScript "openvpn-${name}-up" upScript}";
+      downScriptOption = optionalString (hasDownScript || cfg.updateResolvConf) "down ${pkgs.writeShellScript "openvpn-${name}-down" downScript}";
+
+      authUserPassUnsafeOption = optionalString (cfg.authUserPass != null) "auth-user-pass ${pkgs.writeText "openvpn-credentials-${name}" ''
+            ${cfg.authUserPass.username}
+            ${cfg.authUserPass.password}
+      ''}";
+
+      authUserPassFileOption = optionalString (cfg.authUserPassFile != null) "auth-user-pass ${cfg.authUserPassFile}";
+
+      authUserPassOption = if (authUserPassFileOption != "") then authUserPassFileOption else authUserPassUnsafeOption;
```

Comment:
If you add an assertion that only one of these options can be set (i.e. `cfg.authUserPass -> !cfg.authUserPassFile && cfg.authUserPassFile -> !cfg.authUserPass`) you could just add both `authUserPassUnsafeOption` and `authUserPassFileOption` to the result, as at least one of them will always be an empty string